modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/search/spi/model/index/contributor/DDMStructureModelDocumentContributor.java

```diff
@@ -61,15 +62,18 @@ public void contribute(Document document, DDMStructure ddmStructure) {
 
 			document.addKeyword(Field.STATUS, structureVersion.getStatus());
 			document.addKeyword(Field.VERSION, structureVersion.getVersion());
+
+			document.addKeyword(
+				"resourcePermissionName",
+				_ddmPermissionSupport.getStructureModelResourceName(
+					ddmStructure.getClassNameId()));
 		}
 		catch (PortalException pe) {
 			if (_log.isDebugEnabled()) {
 				_log.debug(pe, pe);
 			}
 		}
 
-		document.addKeyword(
```

Comment:
@rodrigopaulino why did you remove this? DDMTemplateModelDocumentContributor is still indexing this attribute. Any reason to do this?

Reply:
I removed it because it doesn't make sense since DDMStructure doesn't have a resourceClassNameId attribute. This was my first solution, but I saw that after. That's why I sent you this new solution.